src/lib/command/util/util-util.ts

```diff
@@ -29,40 +29,66 @@ export const chooseOptionsDefaults = <T extends object>(): ChooseOptions<T> => (
 	autoChoose: false,
 })
 
-export const chooseOptionsWithDefaults = <T extends object>(options: Partial<ChooseOptions<T>> | undefined): ChooseOptions<T> => ({
+export const chooseOptionsWithDefaults = <T extends object>(
+	options: Partial<ChooseOptions<T>> | undefined,
+): ChooseOptions<T> => ({
 	...chooseOptionsDefaults(),
 	...options,
 })
 
+export type CreateChooseFunctionOptions<T extends object> = {
+	defaultValue: Omit<Required<SelectOptions<T>>['defaultValue'], 'getItem'> & {
+		getItem: (command: APICommand, id: string) => Promise<T>
+	}
+}
+
 export type ChooseFunction<T extends object> = (
 	command: APICommand<SelectFromListFlags>,
 	itemIdOrNameFromArg?: string,
 	options?: Partial<ChooseOptions<T>>) => Promise<string>
 
 export const createChooseFn = <T extends object>(
 	config: SelectFromListConfig<T>,
-	listItems: (client: SmartThingsClient) => Promise<T[]>,
+	listItems: (command: APICommand) => Promise<T[]>,
+	createOptions?: CreateChooseFunctionOptions<T>,
 ): ChooseFunction<T> =>
 	async (
 			command: APICommand<SelectFromListFlags>,
 			itemIdOrNameFromArg?: string,
-			options?: Partial<ChooseOptions<T>>): Promise<string> => {
+			options?: Partial<ChooseOptions<T>>,
+	): Promise<string> => {
 		const opts = chooseOptionsWithDefaults(options)
 
 		// Listing items usually makes an API call which we only want to happen once so we do it
 		// now and just use stub functions that return these items later as needed.
-		const items = await listItems(command.client)
+		const items = await (opts.listItems ?? listItems)(command)
 		const filteredItems = opts.listFilter ? items.filter(opts.listFilter) : items
 		const listItemsWrapper = async (): Promise<T[]> => filteredItems
 
 		const preselectedId = opts.allowIndex
 			? await stringTranslateToId(config, itemIdOrNameFromArg, listItemsWrapper)
 			: itemIdOrNameFromArg
 
-		return selectFromList(command, config, {
+		const selectOptions: SelectOptions<T> = {
 			preselectedId,
 			autoChoose: opts.autoChoose,
 			listItems: listItemsWrapper,
 			promptMessage: opts.promptMessage,
-		})
+		}
+
+		if (opts.useConfigDefault) {
+			if (!createOptions?.defaultValue) {
+				throw Error('invalid state, choose function called with "useConfigDefault" but no default configured')
```

Comment:
I'm not sure if this change would be appropriate for an error message, but it might be easier to parse if it said something like "choose function was called" because it's a little ambiguous whether "choose" is a verb or an adjective here.  (I.e., that "choose function" is a type of function, not an instruction)